src/contextkit-subscriber/property.cpp

```diff
@@ -189,7 +189,9 @@ class UnsubscribeRequest : public Event
         , key_(key)
         , done_(std::move(done))
     {}
-    virtual ~UnsubscribeRequest() {}
+    virtual ~UnsubscribeRequest() {
+        done_.set_value();
+    }
 
     UnsubscribeRequest(SubscribeRequest const&) = delete;
```

Comment:
moreover, deleted copy ctors are not needed anymore, maybe they were left from the prevoius impl.